runelite-client/src/main/java/net/runelite/client/plugins/itemcharges/ItemChargesOverlay.java

```diff
@@ -61,26 +67,27 @@ public Dimension render(Graphics2D graphics)
 	{
 		graphics.setFont(FontManager.getRunescapeSmallFont());
 
-		for (WidgetItem item : getJewelleryWidgetItems())
+		for (WidgetItem widgetItem : getChargeCountWidgetItems())
 		{
-			JewelleryCharges charges = JewelleryCharges.getCharges(item.getId());
+			ItemComposition item = client.getItemDefinition(widgetItem.getId());
+			Integer charges = ChargeCounts.getCharges(item, config);
 
 			if (charges == null)
 			{
 				continue;
 			}
 
-			final Rectangle bounds = item.getCanvasBounds();
+			final Rectangle bounds = widgetItem.getCanvasBounds();
 			final TextComponent textComponent = new TextComponent();
```

Comment:
Probably saving this object during class initialization and then just altering it and rendering might save some memory.

Reply:
I would rather we just use `graphics.drawString` so we don't have to evaluate the regex too.

Reply:
I don't think that is good idea, as one of the purposes of the text component is also consistency of the UI even though that will bring small increase to performance.